src/main/java/it/mulders/mcs/search/SearchCommandHandler.java

```diff
@@ -1,19 +1,25 @@
 package it.mulders.mcs.search;
 
 import it.mulders.mcs.common.Result;
+import it.mulders.mcs.search.printer.DelegatingOutputPrinter;
+import it.mulders.mcs.search.printer.OutputPrinter;
 
 import static it.mulders.mcs.search.Constants.MAX_LIMIT;
 
 public class SearchCommandHandler {
     private final SearchClient searchClient;
-    private final OutputPrinter outputPrinter;
+    private final DelegatingOutputPrinter outputPrinter;
```

Comment:
Now that the `DelegatingOutputPrinter` is immutable again, we no longer need to invoke `setCoordinatePrinter` on it. Does that mean we could rollback the declaration of this instance variable to type `OutputPrinter`?

Reply:
Switched back to `OutputPrinter`. Thank you for make me aware of.